### ISSUE TITLE
Inconsistent Lua formatting

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#11528]
https://github.com/minetest/minetest/commit/80d12dbedb67191a5eb3e4f3c36b04baed1f8afb introduced (unneeded) semicolons to the Lua codebase. We should agree on proper Lua formatting, which should not be enforced by hand, but using a formatter.

**Comment by sfan5** [comment]
Which Lua formatter do you suggest?

**Comment by appgurueu** [comment]
> Which Lua formatter do you suggest?

I'm unfortunately heavily biased towards https://github.com/appgurueu/luafmt but wouldn't recommend it yet.

**Comment by v-rob** [comment]
Sounds like someone has been programming too much C++ :)

**Comment by hecktest** [comment]
We do have a code style, this was simply missed. I use semicolons in my own Lua scripts because it reduces the effort of switching between languages, and provides a few minor benefits.

Just committed a fix. My code style will always be at odds with MT's in both languages and nothing short of auto linting on commit will fix that.

**Comment by Desour** [comment]
> We do have a code style

There's these two wiki pages:
https://dev.minetest.net/Code_style_guidelines
https://dev.minetest.net/Lua_code_style_guidelines

I can't find anything in the lua code style guidelines about semicolons.
(Also imho, unneeded semicolons aren't a big problem. They don't hinder readability, and aren't unacceptably ugly.)

Btw. is it just me or are the `<source>` things on the wiki broken?

**Comment by sfan5** [comment]
> Btw. is it just me or are the `<source>` things on the wiki broken?

Yes, the wiki was recently upgraded. If you know MediaWiki please help fix them.

**Comment by appgurueu** [comment]
Nowadays I think https://github.com/JohnnyMorganz/StyLua would be a good choice for a formatter.
